File: cpp/2.cpp
#include<stdio.h>
#include<math.h>
#include<conio.h>

void main()
{
	int jam, gaji, pajak, asuransi, gajikotor, sisa, jam1, jam2, jamsisa
		, pajak1, pajak2, pajak3, pajak4, gajibersih;
	float bunga1, bunga2, bunga3, bunga4, bunga5, minggu, lembur;

	jam1 = 40;
	jam2 = 20;
	minggu = 1.5;
	lembur = 1.8;
	pajak = 0;
	pajak1 = 100000; bunga1 = 0.05;
	pajak2 = 200000; bunga2 = 0.1;
	pajak3 = 300000; bunga3 = 0.2;
	pajak4 = 600000; bunga4 = 0.3;
					 bunga5 = 0.07;

	printf("penghitungan gaji per minggu\n");

	printf("jumlah jam kerja :");
	scanf("%d", &jam);

	printf("gaji perjam :");
	scanf("%d", &gaji);

	if (jam > 84 && gaji < 0)
	{
		printf("data tidak valid");
		printf("jam kerja melebihi syarat maksimum ");
		printf("gaji tidak boleh minus (negatif)");
	}
	else if (jam <= 84 && gaji < 0)
	{
		printf("data tidak valid");
		printf("gaji tidak boleh minus (negatif)");
	}
	else if (jam > 84 && gaji < 0)
	{
		printf("data tidak valid");
		printf("jam kerja melebihi syarat maksimum ");
	}
	else
	{
		if (jam < 40)
		{
			printf("gaji kotor :\n");
			printf("%d ", jam);
			printf("x ");
			printf("%d ", gaji);
			printf("= ");
			printf("%d\n", jam*gaji);
			gajikotor = jam*gaji;
			printf("gaji kotor = %d\n\n", gajikotor);
		}
		else if (jam > 40 && jam < 60)
		{
			jamsisa = jam - jam1;
			printf("gaji kotor :\n");
			printf("%d ", jam1);
			printf("x ");
			printf("%d ", gaji);
			printf("= ");
			printf("%d\n", jam1*gaji);

			printf("%d ", jamsisa);
			printf("x ");
			printf("%d ", gaji);
			printf("x ");
			printf("%.2f ", minggu);
			printf("= ");
			printf("%.2f\n", jamsisa*gaji*minggu);
			gajikotor = (jam1*gaji) + (jamsisa*gaji*minggu);
			printf("gaji kotor = %d\n\n", gajikotor);
		}
		else if (jam > 40 && jam <= 84)
		{
			jamsisa = jam - jam1 - jam2;
			printf("gaji kotor :\n");
			printf("%d ", jam1);
			printf("x ");
			printf("%d ", gaji);
			printf("= ");
			printf("%d\n", jam1*gaji);

			printf("%d ", jam2);
			printf("x ");
			printf("%d ", gaji);
			printf("x ");
			printf("%.2f ", minggu);
			printf("= ");
			printf("%.2f\n", jam2*gaji*minggu);

			printf("%d ", jamsisa);
			printf("x ");
			printf("%d ", gaji);
			printf("x ");
			printf("%.2f ", lembur);
			printf("= ");
			printf("%.2f\n ", jam2*gaji*lembur);
			gajikotor = (jam1*gaji) + (jam2*gaji*minggu) + (jamsisa*gaji*lembur);
			printf("gaji kotor = %d\n\n", gajikotor);
		}

		if (gajikotor <= 100000)
		{
			printf("penghitungan pajak :\n");
			printf("%d",gajikotor);
			printf(" ");
			printf("tidak kena pajak\n\n");

			printf("biaya asuransi :\n");
			printf("%d ", gajikotor);
			printf("x ");
			printf("%.2f ", bunga5);
			printf("= ");
			asuransi = gajikotor*bunga5;
			printf("%d\n\n", asuransi);

			printf("gaji bersih =");
			printf("%d ", gajikotor);
			printf("- ");
			printf("%d ", pajak);
			printf("- ");
			printf("%d ", asuransi);
			printf("= ");
			gajibersih = gajikotor - pajak - asuransi;
			printf("%d\n", gajibersih);
		}
		else if (gajikotor <= 200000)
		{
			printf("penghitungan pajak :\n");
			sisa = gajikotor - 100000;
			printf("%d", pajak1);
			printf(" ");
			printf("tidak kena pajak\n");
			printf("%d ", sisa);
			printf("x ");
			printf("%.2f ", bunga1);
			printf("= ");
			printf("%.2f ", sisa * bunga1);
			pajak = sisa * bunga1;
			printf("total pajak = %d\n\n", pajak);

			printf("biaya asuransi :\n");
			printf("%.2f ", gajikotor);
			printf("x ");
			printf("%.2f ", bunga5);
			printf("= ");
			asuransi = gajikotor*bunga5;
			printf("%d\n\n", asuransi);

			printf("gaji bersih =");
			printf("%d ", gajikotor);
			printf("- ");
			printf("%d ", pajak);
			printf("- ");
			printf("%d ", asuransi);
			printf("= ");
			gajibersih = gajikotor - pajak - asuransi;
			printf("%d\n", gajibersih);
		}
		else if (gajikotor <= 300000)
		{
			printf("penghitungan pajak :\n");
			sisa = gajikotor - 100000 - 200000;
			printf("%d", pajak1);
			printf(" ");
			printf("tidak kena pajak\n");
			printf("%d ", pajak2);
			printf("x ");
			printf("%.2f ", bunga1);
			printf("= ");
			printf("%.2f\n", pajak2 * bunga1);
			printf("%d ", sisa);
			printf("x ");
			printf("%.2f ", bunga2);
			printf("= ");
			printf("%.2f\n", sisa * bunga2);
			pajak = (pajak2 * bunga1) + (sisa * bunga2);
			printf("total pajak = %d\n\n", pajak);

			printf("biaya asuransi :\n");
			printf("%d ", gajikotor);
			printf("x ");
			printf("%.2f ", bunga5);
			printf("= ");
			asuransi = gajikotor*bunga5;
			printf("%d\n\n", asuransi);

			printf("gaji bersih =");
			printf("%d ", gajikotor);
			printf("- ");
			printf("%d ", pajak);
			printf("- ");
			printf("%d ", asuransi);
			printf("= ");
			gajibersih = gajikotor - pajak - asuransi;
			printf("%d\n", gajibersih);
		}
		else if (gajikotor <= 600000 || gajikotor < 1200000)
		{
			printf("penghitungan pajak :\n");
			sisa = gajikotor - 100000 - 200000 - 300000;
			printf("%d", pajak1);
			printf(" ");
			printf("tidak kena pajak\n");
			printf("%d ", pajak2);
			printf("x ");
			printf("%.2f ", bunga1);
			printf("= ");
			printf("%.2f\n", pajak2 * bunga1);
			printf("%d ", pajak3);
			printf("x ");
			printf("%.2f ", bunga2);
			printf("= ");
			printf("%.2f\n", pajak3 * bunga2);
			printf("%d", sisa);
			printf("x");
			printf("%.2f ", bunga3);
			printf("= ");
			printf("%.2f\n", sisa * bunga3);
			pajak = (pajak2 * bunga1) + (pajak3 * bunga2) + (sisa * bunga3);
			printf("total pajak = %d\n\n", pajak);

			printf("biaya asuransi :\n");
			printf("%d ", gajikotor);
			printf("x ");
			printf("%d ", bunga5);
			printf("= ");
			asuransi = gajikotor*bunga5;
			printf("%d\n\n", asuransi);

			printf("gaji bersih =");
			printf("%d ", gajikotor);
			printf("- ");
			printf("%d ", pajak);
			printf("- ");
			printf("%d ", asuransi);
			printf("= ");
			gajibersih = gajikotor - pajak - asuransi;
			printf("%d\n\n", gajibersih);
		}
		else if (gajikotor > 1200000)
		{
			printf("penghitungan pajak :\n");
			sisa = gajikotor - 100000 - 200000 - 300000 - 600000;
			printf("%d ", pajak1);
			printf(" ");
			printf("tidak kena pajak\n");
			printf("%d ", pajak2);
			printf("x ");
			printf("%.2f ", bunga1);
			printf("= ");
			printf("%.2f\n", pajak2 * bunga1);
			printf("%d ", pajak3);
			printf("x ");
			printf("%.2f ", bunga2);
			printf("= ");
			printf("%.2f\n", pajak3 * bunga2);
			printf("%d ", pajak4);
			printf("x ");
			printf("%.2f ", bunga3);
			printf("= ");
			printf("%.2f\n", pajak4 * bunga3);
			printf("%d ", sisa);
			printf("x ");
			printf("%.2f ", bunga4);
			printf("= ");
			printf("%.2f\n", sisa * bunga4);
			pajak = (pajak2 * bunga1) + (pajak3 * bunga2) + (pajak4 * bunga3) + (sisa * bunga4);
			printf("total pajak = %d\n\n", pajak);

			printf("biaya asuransi :\n");
			printf("%d ", gajikotor);
			printf("x ");
			printf("%.2f ", bunga5);
			printf("= ");
			asuransi = gajikotor*bunga5;
			printf("%d\n\n", asuransi);

			printf("gaji bersih =");
			printf("%d ", gajikotor);
			printf("- ");
			printf("%d ", pajak);
			printf("- ");
			printf("%d ", asuransi);
			printf("= ");
			gajibersih = gajikotor - pajak - asuransi;
			printf("%d\n", gajibersih);
		}
	}
	getch();
	
}

File: csharp/2.cs
using System;

namespace Sample
{
	internal class Program
	{
		private static void Main(string[] args)
		{
			int Jam_Kerja_Normal = 40;
			int Jam_Kerja_Lembur = 60;
			double Bonus_Upah_Liburan = 1.5;
			double Bonus_Upah_Lembur = 1.8;

			double Asuransi_Kesehatan = 0.07;

			double Penghasilan_Tidak_Kena_Pajak = 100_000;
			double Penghasilan_Kena_Pajak_1 = 200_000;
			double Penghasilan_Kena_Pajak_2 = 300_000;
			double Penghasilan_Kena_Pajak_3 = 600_000;

			double Pajak_Penghasilan_1 = 0.05; // 5%
			double Pajak_Penghasilan_2 = 0.1; // 10%
			double Pajak_Penghasilan_3 = 0.2; // 20%
			double Pajak_Penghasilan_4 = 0.3; // 30%

			Console.Write("Total jam kerja: ");
			var Total_Jam_Kerja = int.Parse(Console.ReadLine());

			Console.Write("Upah per jam: ");
			var Upah_per_Jam = int.Parse(Console.ReadLine());

			// cek total jam dan upah per jam valid atau tidak
			// jika tidak sesuai berhentikan program
			if (Total_Jam_Kerja > 84 || Total_Jam_Kerja <= 0)
			{
				// kerja lebih dari 1 minggu
				// 1 minggu = 84 jam
				Console.WriteLine("Total jam kerja hanya boleh 1 hingga 84.");
				return;
			}

			if (Upah_per_Jam <= 0)
			{
				Console.WriteLine("Upah per jam tidak boleh kurang dari 0.");
				return;
			}

			double Gaji_Kotor = 0;

			Console.WriteLine("Gaji Kotor:");

			if (Total_Jam_Kerja > Jam_Kerja_Lembur)
			{
				// waktu kerja > 60 jam 
				Gaji_Kotor = Jam_Kerja_Normal * Upah_per_Jam; // upah normal
				Console.WriteLine($"{ Jam_Kerja_Normal } x { Upah_per_Jam } = { Gaji_Kotor }");

				Gaji_Kotor += 20 * (Upah_per_Jam * Bonus_Upah_Liburan); // upah hari libur
				Console.WriteLine($"20 x { Upah_per_Jam } x { Bonus_Upah_Liburan } = { 20 * (Upah_per_Jam * Bonus_Upah_Liburan) }");

				Gaji_Kotor += (Total_Jam_Kerja - Jam_Kerja_Lembur) * (Upah_per_Jam * Bonus_Upah_Lembur); // upah lembur
				Console.WriteLine($"{ (Total_Jam_Kerja - Jam_Kerja_Lembur) } x { Upah_per_Jam } x { Bonus_Upah_Lembur } = { (Total_Jam_Kerja - Jam_Kerja_Lembur) * (Upah_per_Jam * Bonus_Upah_Lembur) }");
			}
			else if (Total_Jam_Kerja > Jam_Kerja_Normal && Total_Jam_Kerja < Jam_Kerja_Lembur)
			{
				// waktu kerja > 40 jam dan waktu kerja < 60 jam
				Gaji_Kotor = Jam_Kerja_Normal * Upah_per_Jam; // upah normal
				Console.WriteLine($"{ Jam_Kerja_Normal } x { Upah_per_Jam } = { Gaji_Kotor }");

				Gaji_Kotor += (Total_Jam_Kerja - Jam_Kerja_Normal) * (Upah_per_Jam * Bonus_Upah_Liburan); // upah hari libur
				Console.WriteLine($"{ (Total_Jam_Kerja - Jam_Kerja_Normal) } x { Upah_per_Jam } x { Bonus_Upah_Liburan } = { (Total_Jam_Kerja - Jam_Kerja_Normal) * (Upah_per_Jam * Bonus_Upah_Liburan) }");
			}
			else
			{
				// waktu kerja <= 40 jam
				Gaji_Kotor = Total_Jam_Kerja * Upah_per_Jam; // upah normal
				Console.WriteLine($"{ Total_Jam_Kerja } x { Upah_per_Jam } = { Gaji_Kotor }");
			}

			Console.WriteLine($"Gaji kotor: { Gaji_Kotor }");
			Console.WriteLine();

			// hitung pajak
			double Pajak = 0;
			bool Kena_Pajak = true;

			Console.WriteLine("Penghitungan pajak:");

			double temp = 0;

			if (Gaji_Kotor > Penghasilan_Tidak_Kena_Pajak)
			{
				temp = Gaji_Kotor - Penghasilan_Tidak_Kena_Pajak;
				Console.WriteLine($"{ Penghasilan_Tidak_Kena_Pajak } tidak kena pajak.");
			}
			else
			{
				Kena_Pajak = false;
				Console.WriteLine($"{ Gaji_Kotor } tidak kena pajak.");
			}

			if (temp > Penghasilan_Kena_Pajak_1)
			{
				temp -= Penghasilan_Kena_Pajak_1;
				Pajak = Penghasilan_Kena_Pajak_1 * Pajak_Penghasilan_1;
				Console.WriteLine($"{ Penghasilan_Kena_Pajak_1 } x { Pajak_Penghasilan_1 } = { Pajak }");
			}
			else if (Kena_Pajak)
			{
				Pajak = temp * Pajak_Penghasilan_1;
				Console.WriteLine($"{ temp } x { Pajak_Penghasilan_1 } = { Pajak }");
			}

			if (temp > Penghasilan_Kena_Pajak_2)
			{
				temp -= Penghasilan_Kena_Pajak_2;
				Pajak += Penghasilan_Kena_Pajak_2 * Pajak_Penghasilan_2;
				Console.WriteLine($"{ Penghasilan_Kena_Pajak_2 } x { Pajak_Penghasilan_2 } = { Penghasilan_Kena_Pajak_2 * Pajak_Penghasilan_2 }");
			}
			else if (Kena_Pajak)
			{
				Pajak += temp * Pajak_Penghasilan_2;
				Console.WriteLine($"{ temp } x { Pajak_Penghasilan_2 } = { temp * Pajak_Penghasilan_2 }");
			}

			if (temp > Penghasilan_Kena_Pajak_3)
			{
				temp -= Penghasilan_Kena_Pajak_3;
				Pajak += Penghasilan_Kena_Pajak_3 * Pajak_Penghasilan_3;
				Console.WriteLine($"{ Penghasilan_Kena_Pajak_3 } x { Pajak_Penghasilan_3 } = { Penghasilan_Kena_Pajak_3 * Pajak_Penghasilan_3 }");

				// penghasilan yang kena pajak
				// jika masih ada sisa
				if (temp > 0)
				{
					Pajak += temp * Pajak_Penghasilan_4;
					Console.WriteLine($"{ temp } x { Pajak_Penghasilan_4 } = { temp * Pajak_Penghasilan_4 }");
				}
			}
			else if (Kena_Pajak)
			{
				Pajak += temp * Pajak_Penghasilan_3;
				Console.WriteLine($"{ temp } x { Pajak_Penghasilan_3 } = { temp * Pajak_Penghasilan_3 }");
			}

			Console.WriteLine($"Total pajak: {  Pajak }");
			Console.WriteLine();

			var Biaya_Asuransi = Gaji_Kotor * Asuransi_Kesehatan;
			Console.WriteLine($"Biaya asuransi: { Gaji_Kotor } x { Asuransi_Kesehatan } = { Biaya_Asuransi }");

			var Gaji_Bersih = Gaji_Kotor - Pajak - Biaya_Asuransi;
			Console.WriteLine($"Gaji bersih = { Gaji_Bersih }");
		}
	}
}
